istioctl/cmd/injector-list.go

```diff
@@ -0,0 +1,311 @@
+	return cmd
+}
+
+func injectorListCommand() *cobra.Command {
+	var opts clioptions.ControlPlaneOptions
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List sidecar injector and sidecar versions",
+		Long:    `List sidecar injector and sidecar versions`,
+		Example: `  istioctl experimental injector list`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := kubeClientWithRevision(kubeconfig, configContext, opts.Revision)
+			if err != nil {
+				return fmt.Errorf("failed to create k8s client: %v", err)
+			}
+			nslist, err := getNamespaces(client)
+			if err != nil {
+				return err
+			}
+			hooks, err := getHooks(client)
+			if err != nil {
+				return err
+			}
+			pods, err := getPods(client)
+			_ = pods
+			if err != nil {
+				return err
+			}
+			err = printNS(cmd.OutOrStdout(), nslist, hooks, pods)
+			if err != nil {
+				return err
+			}
+			fmt.Println()
+			injectedImages, err := getInjectionImages(client)
+			if err != nil {
+				return err
+			}
+			return printHooks(cmd.OutOrStdout(), nslist, hooks, injectedImages)
+		},
+	}
+
+	return cmd
+}
+
+func getNamespaces(client kube.ExtendedClient) ([]v1.Namespace, error) {
+	nslist, err := client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return []v1.Namespace{}, err
+	}
+	return nslist.Items, nil
+}
+
+func printNS(writer io.Writer, namespaces []v1.Namespace, hooks []admit_v1.MutatingWebhookConfiguration,
+	allPods map[resource.Namespace][]v1.Pod) error {
+	if len(namespaces) == 0 {
+		_, err := fmt.Fprintf(writer, "No namespaces present.\n")
+		return err
+	}
+
+	// TODO sort
+
+	w := new(tabwriter.Writer).Init(writer, 0, 8, 1, ' ', 0)
+	fmt.Fprintln(w, "NAMESPACE\tISTIO-REVISION\tPOD-REVISIONS")
+	for _, namespace := range namespaces {
+		if hideFromOutput(resource.Namespace(namespace.Name)) {
+			continue
+		}
+
+		revision := getInjectedRevision(&namespace, hooks)
+		podcounts := podcountByRevision(allPods[resource.Namespace(namespace.Name)], revision)
+		for injectedRevision, count := range podcounts {
+			fmt.Fprintf(w, "%s\t%s\t%s\n", namespace.Name, revision,
+				fmt.Sprintf("%s: %s", injectedRevision, renderCounts(count)))
+		}
+	}
+	return w.Flush()
+}
+
+func getHooks(client kube.ExtendedClient) ([]admit_v1.MutatingWebhookConfiguration, error) {
+	hooks, err := client.AdmissionregistrationV1().MutatingWebhookConfigurations().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return []admit_v1.MutatingWebhookConfiguration{}, err
+	}
+	return hooks.Items, nil
+}
+
+func printHooks(writer io.Writer, namespaces []v1.Namespace, hooks []admit_v1.MutatingWebhookConfiguration, injectedImages map[string]string) error {
+	if len(hooks) == 0 {
+		_, err := fmt.Fprintf(writer, "No Istio injection hooks present.\n")
+		return err
+	}
+
+	// TODO sort
+
+	w := new(tabwriter.Writer).Init(writer, 0, 8, 1, ' ', 0)
+	fmt.Fprintln(w, "INJECTOR-HOOK\tISTIO-REVISION\tNAMESPACES\tSIDECAR-IMAGE")
```

Comment:
Does it make sense to display namespace in the first column ?
```suggestion
	fmt.Fprintln(w, "NAMESPACES\tINJECTOR-HOOK\tISTIO-REVISION\tSIDECAR-IMAGE")
```